sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -201,18 +202,26 @@ def plot_partial_dependence(
 
         .. versionadded:: 0.22
 
-    kind : {'average', 'individual', 'both'}, default='average'
+    kind : {'average', 'individual', 'both'} or list of such str, \
+            default='average'
         Whether to plot the partial dependence averaged across all the samples
         in the dataset or one line per sample or both.
 
         - ``kind='average'`` results in the traditional PD plot;
         - ``kind='individual'`` results in the ICE plot.
 
+        A list of such strings can be provided. The length of the list should
+        be the same as the number of interaction requested in `features`.
+        This option allows to silent warning when ICE plots are requested for
+        both 1- and 2-way PD.
```

Comment:
As this is only a consequence of the real new feature (i.e. finer grain control), I would leave that out

```suggestion
```